### PR TITLE
Named parameters.

### LINKED ISSUE #52: Unsupported node type ObjectBindingPattern
constructor(bindings:List, {parent=null, defaultBindings=false}={}) {

Error: /Users/alexeagle/Projects/angular/modules/angular2/src/di/injector.js:30:30: Unsupported node type ObjectBindingPattern:  {parent=null, defaultBindings=false}

### PR DESCRIPTION
**Pull request by mprobst**
Fixes #52.

### CODE REVIEW

**Comment by alexeagle** [comment]
Add "fixes #52" to the description

File: test/DartTest.ts [hunk 1]
```diff
@@ -144,6 +144,12 @@ describe('transpile to dart', () => {
       expectTranslate('function x(a = 42) { return 42; }')
           .to.equal(' x ( [ a = 42 ] ) { return 42 ; }');
     });
+    it('supports named parameters', () => {
+      expectTranslate('function x({a, b}) { return a + b; }' +
```
Comment (alexeagle) [inline test/DartTest.ts]: add a test for the syntax with equals sign, like the one in the bug #52

**Comment by mprobst** [comment]
Actually, this is wrong, let me take a second look.

**Comment by mprobst** [comment]
PTAL. Gross hacks have been committed.

**Comment by mprobst** [comment]
@vsavkin @vicb FYI.

File: test/e2e/lib.ts [hunk 2]
```diff
@@ -3,12 +3,12 @@ class MyClass {
 
   MyClass(someVal: string) { this.field = someVal; }
 
-  getField(): string {
+  getField({x = "?"}): string {
```
Comment (alexeagle) [inline test/e2e/lib.ts]: needs merged w/ HEAD (this is now a getter and won't allow the argument...)

File: test/DartTest.ts [hunk 3]
```diff
@@ -145,6 +145,19 @@ describe('transpile to dart', () => {
       expectTranslate('function x(a = 42) { return 42; }')
           .to.equal(' x ( [ a = 42 ] ) { return 42 ; }');
     });
+    it('supports named parameters', () => {
+      expectTranslate('function x({a = "x", b}) { return a + b; }')
+          .to.equal(' x ( { a : "x" , b } ) { return a + b ; }');
+      chai.expect(() => translateSource('function x({a}: number) { return a + b; }'))
+          .to.throw('types on named parameters are unsupported');
+    });
+    it('hacks last object literal paramters into named parameter', () => {
```
Comment (alexeagle) [inline test/DartTest.ts]: spelling parameters

File: test/e2e/lib.ts (hunk 2 above)
Comment (mprobst) [inline test/e2e/lib.ts]: done.

File: test/DartTest.ts (hunk 3 above)
Comment (mprobst) [inline test/DartTest.ts]: done.

**Comment by alexeagle** [comment]
LGTM

**Pull request merged**